examples/archive/bettinggame/src/BettingGame/OffChain.hs

```diff
@@ -0,0 +1,192 @@
+              ),
+            SpendsScript script (GameClose gr0) collectedBetsOutput
+          ]
+            :=>: ( paysPK (operator p) commission :
+                     [paysPK (player b) v | b <- bets, Just v <- [paidToBet commission gr0 bets b]]
+                 )
+        )
+  where
+    isCollectedBets CollectedBets {} _ = True
+    isCollectedBets _ _ = False
+
+    paidToBet :: Pl.Value -> GameResult -> [BetData] -> BetData -> Maybe Pl.Value
+    paidToBet commission gr bets b =
+      let winners = filter ((gr ==) . gameResult) bets
+          total_winners = Pl.sum (map amount winners)
+          total = Pl.sum (map amount bets) Pl.- commission
+       in if gameResult b /= gr
+            then Nothing
+            else
+              Just $
+                Api.unionWith
+                  Builtins.divideInteger
+                  (Api.unionWith (*) total (amount b))
+                  total_winners
+
+-- | Label for 'txClose' skeleton
+data TxClose = TxClose BetParams deriving (Show, Eq)
+
+txReclaim :: MonadBlockChain m => BetParams -> m ()
+txReclaim p = do
+  let script = betValidator p
+  ply <- ownPaymentPubKeyHash
+  betOutputs <- scriptUtxosSuchThat script (isBetOf ply)
+  unless (null betOutputs) $ do
+    -- TODO: Allow to reclaim bets where the amount doesn't
+    -- match the output value
+    let total = Pl.sum [amount d | (_, Bet d) <- betOutputs]
+    void $
+      validateTxConstrLbl
+        TxReclaim
+        ( ( After (collectionDeadline p) :
+            map (SpendsScript script BetReclaim) betOutputs
+          )
+            :=>: [paysPK ply total]
+        )
+  collectedBetsOutputs <- scriptUtxosSuchThat script isCollectedBets
+  case collectedBetsOutputs of
+    collectedBetsOutput@(_, CollectedBets bets) : _ ->
+      void $
+        validateTxConstrLbl
+          TxReclaim
+          ( [ After (publishingDeadline p),
+              SpendsScript script BetReclaim collectedBetsOutput
+            ]
+              :=>: [paysPK (player b) (amount b) | b <- bets]
+          )
+    _ -> return ()
+  where
+    isBetOf ply (Bet d) _ = ply == player d
+    isBetOf _ _ _ = False
+
+    isCollectedBets CollectedBets {} _ = True
+    isCollectedBets _ _ = False
+
+-- | Label for 'txReclaim' skeleton
+data TxReclaim = TxReclaim deriving (Show, Eq)
+
+-- TODO: check deadlines
+
+type BettingGameSchema =
+  C.Endpoint "start" BetParams
+    C..\/ C.Endpoint "bet" (BetParams, (GameResult, Pl.Value))
+    C..\/ C.Endpoint "collectBets" BetParams
+    C..\/ C.Endpoint "close" (BetParams, GameResult)
+    C..\/ C.Endpoint "reclaim" BetParams
+
+mkSchemaDefinitions ''BettingGameSchema
+
+endpoints :: (C.AsContractError e) => C.Promise w BettingGameSchema e ()
+endpoints =
+  C.endpoint @"start" txStart
+    `C.select` C.endpoint @"bet" txBet
+    `C.select` C.endpoint @"collectBets" txCollectBets
+    `C.select` C.endpoint @"close" txClose
+    `C.select` C.endpoint @"reclaim" txReclaim
```

Comment:
I'd move this to a separate module or prefix with a comment explaining this is just an illustration of the interface between cooked and plutus-contract, it is not necessary to do this is you're just looking forward to testing your contract with cooked.

Reply:
Moved to a new module `BettingGame.Endpoint`.